src/BuildScriptGeneratorCli/Commands/Binders/CommandBaseBinder.cs

```diff
@@ -0,0 +1,31 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli.Commands
+{
+    public abstract class CommandBaseBinder<T> : BinderBase<T>
+        where T : CommandBaseProperty
+    {
+#pragma warning disable SA1401 // Fields should be private
```

Comment:
Could you please clarify why we're not making these fields directly private? What would be the issue if we did so?

Reply:
So because it is used later in other binders that inherits from `CommandBaseBinder`, it would be more convenient to go with `protected` than making these private and use getter.

Reply:
Could you define each field as a property instead, like
`protected Argument<string> SourceDir { get; set; }`
or alternatively be set as static and readonly if you would like to keep it as a field?
`protected static readonly Argument<string> sourceDir;`

Reply:
that makes sense. just pushed those changes and waiting for validations. thanks for the suggestion!